lib/rouge/lexers/react.rb

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    load_lexer 'javascript.rb'
+    load_lexer 'xml.rb'
+
+    class ReactJS < Javascript
+      title "ReactJS"
+
+      tag 'jsx'
+      aliases 'jsx'
+      filenames '*.jsx'
+      mimetypes 'text/jsx'
+
+      def self.analyze_text(text)
+        return 1 if text.shebang?('node')
+        return 1 if text.shebang?('jsc')
+        # TODO: rhino, spidermonkey, etc
+      end
+
+      id = /[$a-zA-Z_][a-zA-Z0-9_]*/
+
+      state :root do
```

Comment:
Could you use `prepend` and/or `append`  here to avoid duplicating most of the javascript root state?